src/joint_types.jl

```diff
@@ -131,6 +131,80 @@ function velocity_to_configuration_derivative!(q̇::AbstractVector, jt::Quaterni
     nothing
 end
 
+
+"""
+$(SIGNATURES)
+
+Copy the matrix ``source`` to a block of the same size in ``destination``
+spanning the given given range of rows and columns. In Julia v0.7 and above we
+should be able to eliminate this function in favor of:
+
+```julia
+destination[dest_rows, dest_cols] .= source
+```
+
+since the resulting view will be elided.
+"""
+function _copyto_offset!(destination::AbstractMatrix, source::AbstractMatrix, dest_rows::UnitRange{Int} = 1:size(source, 1), dest_cols::UnitRange{Int} = 1:size(source, 2))
+    @boundscheck length(dest_rows) == size(source, 1)
+    @boundscheck length(dest_cols) == size(source, 2)
```

Comment:
`@boundscheck` doesn't work like that; you still need to throw an error inside the `@boundscheck` expression.